### PR TITLE
CELDEV-1181 - cel-data-renderer rendering off-screen

### PR DESCRIPTION
**Pull request by fpichler**
https://synjira.atlassian.net/browse/CELDEV-1181

### CODE REVIEW

File: src/main/webapp/resources/celDynJS/celData/cel-data-renderer.mjs [hunk 1]
```diff
@@ -66,6 +67,10 @@ export default class CelDataRenderer {
   #insertEntry(entryData) {
     const newElem = this.template.content.cloneNode(true);
     const dataRoot = newElem.querySelector('.cel-data-root');
+    if (this.#renderingFirstElement) {
+      this.htmlElem.replaceChildren();
+      this.#renderingFirstElement = false;
+    }
```
Comment (msladek) [inline src/main/webapp/resources/celDynJS/celData/cel-data-renderer.mjs:70]: this changes the behaviour when reusing the CelDataRenderer. before the children were removed once per `render` call. now they're only removed once per instance.

maybe use a param `#insertEntry(entryData, replaceChildren)` with forEach index instead?
`data.forEach((entryData, idx) => this.#insertEntry(entryData, idx == 0));`

**Review by msladek — CHANGES_REQUESTED** [review]

File: src/main/webapp/resources/celDynJS/celData/cel-data-renderer.mjs (hunk 1 above)
Comment (fpichler) [reply src/main/webapp/resources/celDynJS/celData/cel-data-renderer.mjs:70]: how is paging (rendering in multiple steps) done?

Comment (msladek) [reply src/main/webapp/resources/celDynJS/celData/cel-data-renderer.mjs:70]: changing the page or size attribute calls `render` in the `attributeChangedCallback`. only the actual data returned by the `dataPromise` (the page elements) should be displayed, thus removing the children on each call is part of the previous contract.

**Review by msladek — DISMISSED** [review]

File: src/main/webapp/resources/celDynJS/celData/cel-data-renderer.mjs (hunk 1 above)
Comment (fpichler) [reply src/main/webapp/resources/celDynJS/celData/cel-data-renderer.mjs:70]: if I understand correctly this means that on "load 10 more" the existing elements plus the 10 elements must be rendered. Therefore all existing elements have to be rerendered each time a few additional ones are added. Right?

**Review by msladek — APPROVED** [review]

**Pull request merged**